File: ios/objNative/JSFieldType.m
//
//  JSFieldType.m
//  Supermap
//
//  Created by Yang Shang Long on 2018/9/19.
//  Copyright © 2018年 Facebook. All rights reserved.
//

#import "JSFieldType.h"
#import "SuperMap/FieldType.h"

@implementation JSFieldType
RCT_EXPORT_MODULE();

-(NSDictionary*)constantsToExport{
    return @{
             @"BOOLEAN":@(FT_BOOLEAN),
             @"BYTE":@(FT_BOOLEAN),
             @"INT16":@(FT_INT16),
             @"INT32":@(FT_INT32),
             @"INT64":@(FT_INT64),
             @"SINGLE":@(FT_SINGLE),
             @"DOUBLE":@(FT_DOUBLE),
             @"DATE":@(FT_DATE),
             @"DATETIME":@(FT_DATETIME),
             @"LONGBINARY":@(FT_LONGBINARY),
             @"TEXT":@(FT_TEXT),
             @"CHAR":@(FT_CHAR),
             @"WTEXT":@(FT_WTEXT),
             };
}
@end

File: android/src/main/java/com/supermap/rnsupermap/JSFieldType.java
package com.supermap.rnsupermap;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by Myself on 2017/8/15.
 */

public class JSFieldType extends ReactContextBaseJavaModule {
    public static final String REACT_CLASS="JSFieldType";

    public JSFieldType(ReactApplicationContext context){
        super(context);
    }

    @Override
    public String getName(){return REACT_CLASS;}

    @Override
    public Map<String, Object> getConstants() {
        final Map<String, Object> constants = new HashMap<>();
        constants.put("BOOLEAN", 1);
        constants.put("BYTE", 2);
        constants.put("INT16", 3);
        constants.put("INT32", 4);
        constants.put("INT64", 16);
        constants.put("SINGLE", 6);
        constants.put("DOUBLE", 7);
        constants.put("DATETIME", 23);
        constants.put("LONGBINARY", 11);
        constants.put("TEXT", 10);
        constants.put("CHAR", 18);
        constants.put("WTEXT", 127);
        return constants;
    }
}